specification/ai/Azure.AI.Projects/evaluations/models.tsp

```diff
@@ -30,36 +41,374 @@ model EvaluatorConfiguration {
+  inlineJsonFormat: "inlineJsonFormat",
+
+  string,
+}
+
+@added(Versions.v2025_07_31_preview)
+@doc("Base class for inline evaluation data with format discrimination.")
+@discriminator("dataFormat")
+model InlineData {
+  @doc("Format of the inline data structure")
+  dataFormat: InlineDataFormat;
+}
+
+@doc("Query-response pairs for evaluating Q&A systems and response accuracy.")
+@added(Versions.v2025_07_31_preview)
+model QueryResponseInlineMessages extends InlineData {
+  @doc("Specifies query-response format")
+  dataFormat: InlineDataFormat.queryResponseMessageFormat;
+
+  @doc("Array of query-response pairs with optional context and ground truth")
+  messages: QueryResponseMessage[];
+}
+
+@added(Versions.v2025_07_31_preview)
+@doc("Multi-turn conversations for evaluating dialogue systems and context awareness.")
+model EvaluatorMessages extends InlineData {
+  @doc("Specifies evaluation messages format")
+  dataFormat: InlineDataFormat.evaluationMessagesFormat;
+
+  @doc("Array of messages representing representing queries")
+  query: Message[];
+
+  @doc("Array of messages representing responses")
+  response: Message[];
+
+  @doc("Array of tool definitions that are used in the conversation")
+  toolDefinitions: AgentToolDefinition[];
+}
+
+@added(Versions.v2025_07_31_preview)
+@doc("Custom JSON format for complex evaluation scenarios requiring flexible data structures.")
+model InlineJson extends InlineData {
+  @doc("Specifies JSON format")
+  dataFormat: InlineDataFormat.inlineJsonFormat;
+
+  @doc("Array of JSON strings with custom fields and metadata")
+  messages: string[];
+}
+
+@doc("Data source using inline data provided directly in the request. Size limit is 2 MB")
+@added(Versions.v2025_07_31_preview)
+model InlineDataSource extends EvaluationDataSource {
+  @doc("Specifies inline JSON data source")
+  type: EvaluationDataSourceType.InlineData;
+
+  @doc("Optional unique identifier for the inline data source. This can be an agent id or a custom identifier to distinguish between different inline data sources.")
+  id?: string;
+
+  @doc("Inline data structured according to the specified format")
+  @visibility(Lifecycle.Create)
+  data?: InlineData;
+
+  @doc("Dataset id for the uploaded inline data.")
+  @visibility(Lifecycle.Read)
+  inlineDatasetId: DatasetId;
+}
+
+@doc("Data source that uses a dataset registered and stored in AI Foundry workspace. This is the recommended approach for large datasets or reusable evaluation data.")
+@added(Versions.v2025_07_31_preview)
+model FoundryDatasetDataSource extends EvaluationDataSource {
+  @doc("Specifies that this data source uses a registered dataset")
+  type: EvaluationDataSourceType.Dataset;
+
+  @doc("Unique identifier of the dataset registered in AI Foundry workspace")
+  datasetId: DatasetId;
+}
+
+@doc("Common properties shared across model deployment configurations used in evaluations.")
+@added(Versions.v2025_07_31_preview)
+model EvaluationModelSourceCommon {
```

Comment:
If this is just a bag of properties, it should be an alias. This is presumably not a type we actually want to show up in the resulting api/client libraries/docs.